lib/actions/field-trip.js

```diff
@@ -695,7 +695,9 @@ function checkValidityAndCapacity(state, request) {
       itinerary.legs
         .filter((leg) => isTransit(leg.mode))
         .forEach((leg) => {
-          const { tripId } = leg
+          let { tripId } = leg
+          // OTP2 support
+          if (!tripId) tripId = leg?.trip?.id
```

Comment:
Just use the OTP2 mode always! OTP1 support is gone so there's no point in keeping legacy stuff around.